### PR TITLE
SeqScan completed

### PR DESCRIPTION
**Pull request by korbiniak**

### CODE REVIEW

File: komfydb/storage/heap_page.cc [hunk 1]
```diff
@@ -14,34 +17,51 @@ absl::StatusOr<bool> TuplePresent(std::vector<uint8_t>& header, int i) {
...
   data_idx += Type::INT_LEN;
   return field;
 }
 
 std::unique_ptr<StringField> ParseString(std::vector<uint8_t>& data,
                                          int& data_idx) {
+  // TODO This function does not care about the length of string encoded
+  // in the data file and we parse the string until the first null byte.
+  // Should we leave it like this or read the amount of data specified by
+  // the integer before the string?
```
Comment (kacper776) [inline komfydb/storage/heap_page.cc:37]: I think - read only <length> bytes? That would make more sense if we have it on disk.

File: komfydb/storage/heap_page.cc [hunk 2]
```diff
@@ -61,15 +81,20 @@ absl::StatusOr<std::unique_ptr<HeapPage>> HeapPage::Create(
...
-    for (int j = 0; j < n_fields; j++) {
-      ASSIGN_OR_RETURN(Type field_type, td->GetFieldType(j));
-
-      if (field_type.GetValue() == Type::INT) {
-        RETURN_IF_ERROR(record.SetField(j, ParseInt(data, data_idx)));
-      } else if (field_type.GetValue() == Type::STRING) {
-        RETURN_IF_ERROR(record.SetField(j, ParseString(data, data_idx)));
+    if (!TuplePresent(header, i).value()) {
+      data_idx += td->GetSize();
+    } else {
```
Comment (kacper776) [inline komfydb/storage/heap_page.cc:86]: I would prefer `continue` over big `else` statement.

Comment (korbiniak) [reply komfydb/storage/heap_page.cc:86]: Good point

File: komfydb/storage/heap_page.cc (hunk 1 above)
Comment (korbiniak) [reply komfydb/storage/heap_page.cc:37]: Wdym "read only bytes"?

Comment (kacper776) [reply komfydb/storage/heap_page.cc:37]: "read the amount of data specified by the integer before the string"

**Pull request merged**